Improve maxspeed/freespeed and lanes extraction

dts-ait opened the pull request:
- both now also consider :forward and :backward tags
- maxspeed supports values in kph (default), mph, knots
- maxspeed supports the special values 'none' and 'walk'
- maxspeed always uses the first value if several values are given (semicolon-separated) -> removed the config option "guessFreeSpeed", which is no longer necessary because the parser handles multiple values properly
- lanes calculation respects bus, psv,.. lanes (as described in the OSM wiki)
- fixed lanes calculation on bidirectional ways (lanes-value was not divided by 2)
- many unit tests and a new .osm file with test streets

polettif commented:
Thank you, these are great improvements! The whole speed and lane conversion was too messy, it's way clearer now. I especially appreciate the unit tests and new test data set.

Two things: I wondered about the maxspeed=walk tag. Turns out it's a proposal and concerns cars not pedestrians as I initially assumed. Maybe add a short comment with an osm wiki link in the code to make this clear. And you can remove the "minor" highway defaults, no idea how that ended up there. We're ready to merge after these changes.

dts-ait commented:
Thank you for the feedback and the appreciation. I removed the reference to highway=minor. Regarding maxspeed=walk: I already added a link to the relevant OSM wiki page in the class documentation of OsmMultimodalNetworkConverter, is this sufficient?

polettif commented:
> I already added a link to the relevant OSM wiki page in the class documentation of OsmMultimodalNetworkConverter, is this sufficient?
Ah sorry, didn't see it there, I just looked at the comment for maxspeed=none for Germany. Class documentation is even better.

dts-ait commented:
I was thinking more about the 200kph I chose as value for maxspeed=none on German motorways. Maybe we should use a lower limit since we use the maxspeed to extract an approximation of the *free speed* - and I guess not many cars will drive 200kph. What is a sensible value though? Something between 120 and 140?

jfbischoff commented:
We tend to use something between 100 and 120 kph, but it also depends a bit on the scope of the model.

polettif commented:
Then we should probably move SPEED_LIMIT_WALK_KPH and SPEED_LIMIT_NONE_KPH to the config, with adjusted default values.
